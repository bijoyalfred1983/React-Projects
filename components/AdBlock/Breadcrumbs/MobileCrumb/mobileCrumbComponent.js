/* @flow */
import React, { PureComponent } from 'react';
import { Breadcrumb } from 'semantic-ui-react';
import * as styles from '../styles.scss';

type Props = {
  data: any
};

type State = {
  showFullCrumb: boolean,
  crumbData: any
};

class MobileCrumb extends PureComponent<Props, State> {
  constructor(props: any) {
    super(props);
    this.state = {
      showFullCrumb: false,
      crumbData: this.props.data
    };
  }

  componentWillReceiveProps(nextProps) {
    const newData = nextProps.data;
    this.setState({
      crumbData: newData
    });
  }

  onClickEllipsis = () => {
    this.setState({ showFullCrumb: true });
  };

  render() {
    // console.log('breadcrumbdata', this.state.crumbData);
    let breadcrumbTag;
    if (
      this.state.crumbData &&
      this.state.crumbData.length >= 3 &&
      !this.state.showFullCrumb
    ) {
      breadcrumbTag = (
        <span>
          <Breadcrumb.Section className={styles.homeLink}>
            <button onClick={this.onClickEllipsis} className={styles.homeLink}>
              {`... `}
            </button>
          </Breadcrumb.Section>
          <Breadcrumb.Divider className={styles.dividerLink} />
        </span>
      );
    }

    return (
      <Breadcrumb className={styles.breadcrumbLinkAZ}>
        {breadcrumbTag}
        {this.state.crumbData &&
          this.state.crumbData.map((breadcrumbItem, index) => {
            if (
              !this.state.showFullCrumb &&
              index <= this.state.crumbData.length - 3
            ) {
              return false;
            }

            if (
              parseInt(index, 10) ===
              parseInt(this.state.crumbData.length - 1, 10)
            ) {
              return (
                <Breadcrumb.Section
                  key={breadcrumbItem.displayName}
                  active
                  className={styles.activeLink}
                >
                  {' '}
                  {breadcrumbItem.displayName}{' '}
                </Breadcrumb.Section>
              );
            }
            return (
              <span key={breadcrumbItem.displayName}>
                <Breadcrumb.Section className={styles.homeLink}>
                  <a
                    href={breadcrumbItem.url || '/'}
                    className={styles.homeLink}
                  >
                    {' '}
                    {breadcrumbItem.displayName}{' '}
                  </a>
                </Breadcrumb.Section>
                <Breadcrumb.Divider className={styles.dividerLink} />
              </span>
            );
          })}
      </Breadcrumb>
    );
  }
}

export default MobileCrumb;
