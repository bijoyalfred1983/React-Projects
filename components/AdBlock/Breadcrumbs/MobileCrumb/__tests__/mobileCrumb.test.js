import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import MobileCrumb from '../mobileCrumbComponent';

describe('<MobileCrumb />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      data: [
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Parts',
          url: '/Parts'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Brakes And Traction Control',
          url: '/parts/brakes-and-traction-control'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Disc Brake System',
          url: '/parts/brakes-and-traction-control/disc-brake-system'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Brake Pads',
          url: '/brakes-and-traction-control/brake-pads'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Duralast PlatinumBattery_BT',
          url: ''
        }
      ]
    };
  });

  test('renders on mobile correctly', () => {
    const wrapper = mount(<MobileCrumb {...PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('on click of ellipsis show expanded view', () => {
    const wrapper = mount(<MobileCrumb {...PROPS} />);
    wrapper.instance().onClickEllipsis();
  });
});
