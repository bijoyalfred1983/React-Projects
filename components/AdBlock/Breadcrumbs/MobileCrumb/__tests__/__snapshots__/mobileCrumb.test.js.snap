// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileCrumb /> renders on mobile correctly 1`] = `
<MobileCrumb
  data={
    Array [
      Object {
        "@class": "com.autozone.diy.vo.AZBreadcrumbVO",
        "displayName": "Parts",
        "url": "/Parts",
      },
      Object {
        "@class": "com.autozone.diy.vo.AZBreadcrumbVO",
        "displayName": "Brakes And Traction Control",
        "url": "/parts/brakes-and-traction-control",
      },
      Object {
        "@class": "com.autozone.diy.vo.AZBreadcrumbVO",
        "displayName": "Disc Brake System",
        "url": "/parts/brakes-and-traction-control/disc-brake-system",
      },
      Object {
        "@class": "com.autozone.diy.vo.AZBreadcrumbVO",
        "displayName": "Brake Pads",
        "url": "/brakes-and-traction-control/brake-pads",
      },
      Object {
        "@class": "com.autozone.diy.vo.AZBreadcrumbVO",
        "displayName": "Duralast PlatinumBattery_BT",
        "url": "",
      },
    ]
  }
>
  <Breadcrumb>
    <div
      className="ui breadcrumb"
    >
      <span>
        <BreadcrumbSection>
          <div
            className="section"
            onClick={[Function]}
          >
            <button
              onClick={[Function]}
            >
              ... 
            </button>
          </div>
        </BreadcrumbSection>
        <BreadcrumbDivider>
          <div
            className="divider"
          >
            /
          </div>
        </BreadcrumbDivider>
      </span>
      <span
        key="Brake Pads"
      >
        <BreadcrumbSection>
          <div
            className="section"
            onClick={[Function]}
          >
            <a
              href="/brakes-and-traction-control/brake-pads"
            >
               
              Brake Pads
               
            </a>
          </div>
        </BreadcrumbSection>
        <BreadcrumbDivider>
          <div
            className="divider"
          >
            /
          </div>
        </BreadcrumbDivider>
      </span>
      <BreadcrumbSection
        active={true}
        key="Duralast PlatinumBattery_BT"
      >
        <div
          className="active section"
          onClick={[Function]}
        >
           
          Duralast PlatinumBattery_BT
           
        </div>
      </BreadcrumbSection>
    </div>
  </Breadcrumb>
</MobileCrumb>
`;
